crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -956,11 +968,32 @@ pub fn is_const(expr: &Expression) -> bool {
     expr.mul_terms.is_empty() && expr.linear_combinations.is_empty()
 }
 
+pub fn mul_with_witness(evaluator: &mut Evaluator, a: &Expression, b: &Expression) -> Expression {
+    let a_arith;
+    let a2 = if !a.mul_terms.is_empty() {
```

Comment:
Instead of `let a2 = `  can you change this to `let a_arith = ` 

Then `mul(a2, b2)` can be changed to `mul(&a_arith, &b_arith)`

Reply:
Done